src/authorization/AuxiliaryAuthorizer.ts

```diff
@@ -0,0 +1,55 @@
+import type { AuxiliaryManager } from '../ldp/auxiliary/AuxiliaryManager';
+import { getLoggerFor } from '../logging/LogUtil';
+import { NotImplementedHttpError } from '../util/errors/NotImplementedHttpError';
+import type { AuthorizerArgs } from './Authorizer';
+import { Authorizer } from './Authorizer';
+
+/**
+ * An authorizer for auxiliary resources.
+ * Calls the source authorizer with potentially updated permissions
+ * and the associated resource identifier.
+ */
+export class AuxiliaryAuthorizer extends Authorizer {
+  protected readonly logger = getLoggerFor(this);
+
+  private readonly authorizer: Authorizer;
+  private readonly manager: AuxiliaryManager;
+
+  public constructor(authorizer: Authorizer, manager: AuxiliaryManager) {
```

Comment:
If you think it helps clarity (also for local var).

```suggestion
  public constructor(resourceAuthorizer: Authorizer, manager: AuxiliaryManager) {
```